docs: Link data templates to index function (#1843)
gohugoio/hugoDocs: content/en/templates/data-templates.md

Note the use of the [`markdownify` template function][markdownify]. This will send the description through the Markdown rendering engine.

> ## Use data in front matter
> 
> You might want to access different properties of your data in front matter.
> For example, you want to specify a `location` for a page with specific information about that location.
> 
> To do so, use the [`index` function][index].

Reviewer: That someone was me, thank you for providing the solution, @CollierCZ !
I will take the liberty of suggesting a change more specific to what the root cause of my problem was:
```suggestion
### Named values in variables

You might want to access the key name as a variable. To do so, use the [`index` function][index], with the variable as the second parameter.

{{< highlight plaintext >}}
{{ $keyName := "Name" }}
{{ index .Site.Data.User0123 $keyName }}
{{< /highlight >}}

If the `Name` key was an object, you could also access its properties:

{{< highlight plaintext >}}
{{ $keyName := "Name" }}
{{ (index .Site.Data.User0123 $keyName).propertyName }}
{{< /highlight >}}
```

I had to suggest the highlight shortcode here, using three backticks seems to collide with Github's suggestions.

Author: You can get around that on GitHub by using four backticks for the suggestion itself

Author: ````suggestion
## Named values in variables

You might want to access a key name as a variable.
To do so, use the [`index` function][index] with the variable as the second parameter.
```plaintext
{{ $keyName := "Name" }}
{{ index .Site.Data.User0123 $keyName }}
```

If the `Name` key is an object, you can also access its properties:

```plaintext
{{ $keyName := "Name" }}
{{ (index .Site.Data.User0123 $keyName).propertyName }}
```
````

Note the use of the [`markdownify` template function][markdownify]. This will send the description through the Markdown rendering engine.

## Keys as variables

> You might want to access a key name as a variable.
> To do so, use the [`index` function][index] with the variable as the second parameter.
> ```plaintext

Reviewer 2: Markdown style suggestion to match the rest of the file... (Seems like they're not using language names on their fenced code blocks, either, which seems odd. I like that you provided one, although I bet calling these two `go` would get nice highlighting. :) )

````suggestion
You might want to access a key name as a variable. To do so, use the [`index` function][index] with the variable as the second parameter.

```plaintext
````

Reviewer 2: Also: need to define the link target for `[index]` at the bottom of the file with the others.
